File: Task#2/Python/Enumeration_Of_Divisors.py
import math

# Функция перебора делителей
def dividers(num):
    k = 0
    if num == 1:
        print("Not prime, not composite")
        return 0
    for i in range(2, int(math.sqrt(num)) + 1):
        if num % i == 0:
            k += 1
    if k == 0:
        print("Is prime")
    else:
        print("Not prime")
    return 0

num = int(input("Enter the number: "))
dividers(num)

File: Task#2/C++/Enumeration_Of_Divisors.cpp
#include <iostream>
#include <math.h>
using namespace std;

// Функция перебора делителей
int dividers(int num) {
	int k = 0;
	if (num == 1) {
		cout << "Not prime, not composite!\n" << endl;
		return 0;
	}
	for (int i = 2; i <= sqrt(num); i++) {
		if (num % i == 0) {
			k += 1;
		}
	}
	if (k == 0) {
		cout << "Is prime!\n" << endl;
	}
	else {
		cout << "Not prime!\n" << endl;
	}
	return 0;
}

int main() {
	int num;
	cout << "Enter the number: ";
	cin >> num;
	dividers(num);
	return 0;
}
